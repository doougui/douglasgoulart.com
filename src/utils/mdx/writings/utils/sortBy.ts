import { Writing } from 'types/Writing';
import { parseToDate } from 'utils/time/parseToDate';
import { SortTypesKeys } from '../types';

function byNew(data: Writing[]) {
  return data.sort((a, b) => {
    const aAsDate = parseToDate(a.publishedAt);
    const bAsDate = parseToDate(b.publishedAt);
    return Number(bAsDate) - Number(aAsDate);
  });
}

function byName(data: Writing[]) {
  return data.sort((a, b) => (a.title > b.title ? 1 : -1));
}

export function sortBy(data: Writing[], type: SortTypesKeys): Writing[] {
  const sortTypes: { [key in SortTypesKeys]: () => Writing[] } = {
    new: () => byNew(data),
    name: () => byName(data),
  };

  return sortTypes[type]();
}
