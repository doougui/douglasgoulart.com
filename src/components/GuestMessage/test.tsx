import { formatDistance } from 'date-fns';
import { render, screen } from 'utils/tests';

import { GuestMessage, GuestMessageProps } from '.';

const props: GuestMessageProps = {
  message: 'Nice work',
  author: 'Douglas Pinheiro Goulart',
  date: new Date(2002, 8, 22),
};

describe('<GuestMessage />', () => {
  it('should render message, author and date', () => {
    const { container } = render(<GuestMessage {...props} />);

    expect(screen.getByText('Nice work')).toBeInTheDocument();
    expect(screen.getByText('Douglas Pinheiro Goulart')).toBeInTheDocument();
    expect(
      screen.getByText(
        formatDistance(props.date, new Date(), { addSuffix: true }),
      ),
    ).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
