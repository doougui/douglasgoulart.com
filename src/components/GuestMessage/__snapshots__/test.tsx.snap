// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GuestMessage /> should render message, author and date 1`] = `
.c0 {
  font-size: 1.6rem;
  line-height: 2.4rem;
  color: #EDEDEF;
}

.c1 {
  font-size: 1.6rem;
  line-height: 2.4rem;
  margin-top: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  color: #888888;
}

.c3 {
  color: #504F57;
}

<div
  class=""
>
  <p
    class="c0"
  >
    Nice work
  </p>
  <div
    class="c1"
  >
    <p
      class="c2"
    >
      Douglas Pinheiro Goulart
    </p>
    <p
      class="c3"
    >
      almost 20 years ago
    </p>
  </div>
</div>
`;
